### ISSUE TITLE
[NOTICE] VERSIONS 2.1.0 & 2.2.0 broke CommonJs compatibility

### ISSUE THREAD

**Issue by dryajov** [dryajov/promisify-this#2]
# VERSIONS 2.1.0 & 2.2.0 broke CommonJs compatibility

## TL;DR

> Either upgrade to version 3.0.0 **and** use a named export, as in:
>
> `import { promisify } from 'promisify-this'`
>
> or
>
> `const { promisify } = require('promisify-this')`
>
> or downgrade to version 2.0.2 to continue using the previous CommonJs default export pattern of:
>
> `const promisify = require('promisify-this)`
>
> Alternatively, you can continue using versions 2.1.0 & 2.2.0 (*note however, that this versions have been deprecated*) with the following syntax:
>
> `const {default: promisify} = require('promisify-this')`

## Typescript migration and breaking changes

This module has been rewritten in Typescript beginning with version 2.1.0. The previous pure js implementation relied on the CommonJs pattern of `module.exports = function(){}`. This pattern doesn't work well under the new ES6 modules spec, nor is it fully supported by Typescript as such. Versions `2.1.0` and `2.2.0` relied on the `default export` syntax which doesn't simply map to `module.exports = function(){}`, but adds a new entry - `default` in the exports object, this in turn, ends up mapping to `module.exports = { default: function(){} }` in CommonJs, which is not the intent and breaks compatibility.

Beginning with version `3.0.0` the `default export` has been removed in favor on named exports.
